File: Chapter03/Java/TestResources/app/src/main/java/com/ai/android/book/resources/MonitoredActivity.java
package com.ai.android.book.resources;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.res.Configuration;
import android.os.Bundle;
import android.util.Log;

public class MonitoredActivity extends AppCompatActivity {

    private String tag = null;

    protected MonitoredActivity(String intag) {
        tag = intag;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d(tag, "onCreate.");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(tag, "onPause. I may be partially or fully invisible");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(tag, "onStop. I am fully invisible");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(tag, "onDestroy. about to be removed");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(tag, "onRestart. UI controls are there.");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(tag, "onStart. UI may be partially visible.");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(tag, "onResume. UI fully visible.");
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        Log.d(tag, "onRestoreInstanceState. You should restore state");
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        Log.d(tag, "onSaveInstanceState. Yout should load up the bundle");
    }

    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        Log.d(tag, "onConfigurationChanged.");
    }
}

File: Chapter03/Kotlin/TestResources/app/src/main/java/com/ai/android/book/resources/MonitoredActivity.kt
package com.ai.android.book.resources

import android.content.res.Configuration
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

open class MonitoredActivity(): AppCompatActivity() {

    private var tag: String? = null

    constructor(intag: String): this() {
        this.tag = intag
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onPause() {
        super.onPause()
        Log.d(tag, "onPause. I may be partially or fully invisible");
    }

    override fun onStop() {
        super.onStop()
        Log.d(tag, "onStop. I am fully invisible");
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(tag, "onDestroy. about to be removed");
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(tag, "onRestart. UI controls are there.");
    }

    override fun onStart() {
        super.onStart()
        Log.d(tag, "onStart. UI may be partially visible.");
    }

    override fun onResume() {
        super.onResume()
        Log.d(tag, "onResume. UI fully visible.");
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        Log.d(tag, "onRestoreInstanceState. You should restore state");
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        Log.d(tag, "onSaveInstanceState. You should load up the bundle");
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        Log.d(tag, "onConfigurationChanged.");
    }
}